src/base/scene.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <memory>
+
+#include "camera.h"
+#include "mesh.h"
+#include "object.h"
+#include "tiny_gltf.h"
+
+namespace gm {
+class Scene {
+ public:
+  Scene(){};
+  Scene(const std::string &filepath);
+  void addObject(std::shared_ptr<SceneObject> o);
+  std::shared_ptr<PerspectiveCamera> getCamera();
+  std::vector<std::shared_ptr<SceneObject>> objects;
+
+ private:
+  std::shared_ptr<PerspectiveCamera> load_camera(tinygltf::Node node,
```

Comment:
Check the parameters for each function in this class and see if they can be made into const&. Otherwise we are doing too many pass-by-value calls here.

Reply:
Fixed in 3605540